standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/parser/Filter.g

```diff
@@ -52,34 +53,36 @@ import java.util.regex.Pattern;
   private static final Pattern datePattern = Pattern.compile(".*(\\d\\d\\d\\d-\\d\\d-\\d\\d).*");
   private static final Pattern timestampPattern =
       Pattern.compile(".*(\\d\\d\\d\\d-\\d\\d-\\d\\d \\d\\d:\\d\\d:\\d\\d).*");
-  private static final ThreadLocal<SimpleDateFormat> dateFormat =
-       new ThreadLocal<SimpleDateFormat>() {
-    @Override
-    protected SimpleDateFormat initialValue() {
-      SimpleDateFormat val = new SimpleDateFormat("yyyy-MM-dd");
-      val.setLenient(false); // Without this, 2020-20-20 becomes 2021-08-20.
-      val.setTimeZone(TimeZone.getTimeZone("UTC"));
-      return val;
-    };
-  };
-  private static final ThreadLocal<DateTimeFormatter> timestampFormat =
-       new ThreadLocal<DateTimeFormatter>() {
-    @Override
-    protected DateTimeFormatter initialValue() {
-      DateTimeFormatter val = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
-          .withZone(TimeZone.getTimeZone("UTC").toZoneId());
-      return val;
-    };
-  };
 
-  public static java.sql.Date extractDate(String input) {
+  private static final ThreadLocal<DateTimeFormatter> dateFormat = createDateTimeFormatter("yyyy-MM-dd");
+  private static final ThreadLocal<DateTimeFormatter> timestampFormat = createDateTimeFormatter("yyyy-MM-dd HH:mm:ss");
+
+  private static ThreadLocal<DateTimeFormatter> createDateTimeFormatter(String format) {
+    return new ThreadLocal<DateTimeFormatter>() {
+               @Override
+               protected DateTimeFormatter initialValue() {
+                 DateTimeFormatter val = DateTimeFormatter.ofPattern(format)
+                     .withZone(TimeZone.getTimeZone("UTC").toZoneId());
```

Comment:
Since we are using `LocalDate` and `LocalDateTime` I don't know how much sense it makes to use a zone here. However, it was like that before so we can keep it as is.